test/index.test.js

```diff
@@ -208,6 +208,105 @@ describe('stylelint-webpack-plugin', function () {
     });
   });
 
+  context('different values for the `emitErrors` flag', function () {
+    it('emits errors as warnings when asked to', function () {
+      var config = {
+        context: './test/fixtures/single-error',
+        entry: './index',
+        plugins: [
+          new StyleLintPlugin({
+            configFile: configFilePath,
+            quiet: true,
+            emitErrors: false
+          })
+        ]
+      };
+
+      return pack(assign({}, baseConfig, config))
+        .then(function (stats) {
+          expect(stats.compilation.errors).to.have.length(0);
+          expect(stats.compilation.warnings).to.have.length(1);
+        });
+    });
+
+    it('still indicates that errors are errors to the user, even when emitting them as warnings', function () {
+      var config = {
+        context: './test/fixtures/single-error',
+        entry: './index',
+        plugins: [
+          new StyleLintPlugin({
+            configFile: configFilePath,
+            quiet: true,
+            emitErrors: false
+          })
+        ]
+      };
+
+      return pack(assign({}, baseConfig, config))
+        .then(function (stats) {
+          expect(stats.compilation.warnings[0]).to.contain('✖');
+        });
+    });
+
+    it('still indicates that warnings are warnings to the user, even when emitting errors as warnings too', function () {
+      var config = {
+        context: './test/fixtures/rule-warning',
+        entry: './index',
+        plugins: [
+          new StyleLintPlugin({
+            configFile: configFilePath,
+            quiet: true,
+            emitErrors: false
+          })
+        ]
+      };
+
+      return pack(assign({}, baseConfig, config))
+        .then(function (stats) {
+          expect(stats.compilation.warnings[0]).to.contain('⚠');
+        });
+    });
+
+    it('does not emit errors as warnings when asked not to', function () {
+      var config = {
+        context: './test/fixtures/single-error',
+        entry: './index',
+        plugins: [
+          new StyleLintPlugin({
+            configFile: configFilePath,
+            quiet: true,
+            emitErrors: true
+          })
+        ]
+      };
+
+      return pack(assign({}, baseConfig, config))
+        .then(function (stats) {
+          expect(stats.compilation.errors).to.have.length(1);
+          expect(stats.compilation.warnings).to.have.length(0);
+        });
+    });
+
+    it('does not emit errors as warnings when `emitErrors` is not provided', function () {
```

Comment:
Yes, it is at the moment. I added it with the idea of atomic unit tests, each testing a specific property of the code. That way, if the expected behaviour of the other unit test changes, this property will still be tested.